### ISSUE TITLE
Cherenkov photons wavelength range 

### ISSUE THREAD

**Issue by jsitarek** [cta-observatory/lst-sim-config#17]
Hi @moralejo @Voutsi 

sorry for pointing this out only now, maybe it is too late to tweak with this parameter, but I'm wondering if the wavelength range of the Cherenkov photons was optimized for this production?
The one that is in the Corsika input cards is 240 - 900 nm. The QE of LST (and MAGIC PMTs) only reaches 700 nm or so (for LST it is 0.2% at 700nm) I guess it could be cut easily to 700 or 750 nm if you want to be extra secure. 
For low wavelengths it is also probably too conservative. The LST plexiglass at 270 nm has a transmission of 0.05% (and of course the atmospheric transmission is also huge). 

Since the emitted spectrum is ~lambda^-2 if we skip the 240-270 nm part we would save 15% of Corsika file sizes and processing time, while skipping 700-900 (750-900) nm would save another 10% (7%).

**Comment by moralejo** [comment]
Hi @jsitarek, no, actually we did not discuss at all about the wavelength range. Good point indeed. 

I don't see a big problem with changing that for the next phases of the production, i.e. the next declination lines for the training sample, and for the whole test sample.

I wonder how complicated it would be to "trim" the already produced files, so that they are lighter and identical in settings... :)

**Comment by maxnoe** [comment]
> I wonder how complicated it would be to "trim" the already produced files, so that they are lighter and identical in settings... :)

You keep coming up with reasons why we should implement the O part of EventIO into `pyeventio`.... :roll_eyes:

**Comment by jsitarek** [comment]
modifying existing files might be pretty tricky, because we do not store individual photons, but bunches of them (and I guess the wavelength is simulated later on), so you either would need to make the bunch size smaller which would not really help and would end up with a fractional number of photons, or skip every X-th bunch, which would increase (probably very little) the fluctuation

**Comment by moralejo** [comment]
I think the bunches are treated like "single photons". That is, all of them have the same wavelength (and produce a p.e. or not based on it). So that is probably not an issue. But probably not worth to waste time on that, won't make an impact on the total production size if we change it before we move to the next pointings.

**Comment by maxnoe** [comment]
> and I guess the wavelength is simulated later on)

It depends on the settings. If the IACT extension is configured to make the cherenkov angle wavelength dependent (the option is called `CERWLEN` I think), the wavelength is already drawn at that point. That also implies a bunchsize of 1, if I remember correctly. 

EDIT: yes

> CERWLEN selects Cherenkov code including the wavelength dependent generation of Cherenkov photons and respects their wavelength dependent refraction of the atmosphere. This
option is only available in connection with the CERENKOV option.

**Comment by jsitarek** [comment]
@maxnoe , @moralejo 
indeed, but I think we are not using CERWLEN option, so the standard bunches have only number of photons, without the wavelength.  And this CERWLEN option is also tricky, you need much smaller bunch sizes

**Comment by moralejo** [comment]
Still, the files space saving you calculated at low lambda assumes intensity ~ 1/lambda^2,  i.e. disregarding atmospheric absorption.  ~~In reality it is probably negligible. Then, absorption is evaluated just upon photon bunch creation, right? So the savings in computing will likely be small as well. I am wondering now if it is worth changing it now that the production has started.~~

**Comment by jsitarek** [comment]
the absorption is taken into account later in sim_telarray and not in corsika. You should have less bunches with smaller wavelenth range so I think those estimates that I did should still hold

**Comment by maxnoe** [comment]
> Then, absorption is evaluated just upon photon bunch creation, right? 

This is also settings dependent. You can give already absortion coeffcients to CORSIKA (or rather the IACT extension), which will then apply them already in CORSIKA:

> CEFFIC
> selects code to respect the atmospheric absorption, mirror reflectivity, and photo-
> multiplier quantum efficiency of Cherenkov light. This option is only available in
> connection with the CERENKOV option.

You can identify these by looking at the wavelength in the files. If they are negative, these are "photo-electrons" with these efficiencies already applied. I think this setting was used for CTA simulations, but I am not sure...

**Comment by jsitarek** [comment]
@maxnoe I do not think this is how the current LST+MAGIC production is done. The atmosphere absorption options, reflectivities and QE are specified in sim_telarray settings not in Corsika

**Comment by maxnoe** [comment]
@jsitarek Yes, you are right (at least the current configs in the `corsika` branch don't have `CEFFIC`).

But since you were concerned about the CORSIKA file size, I thought I'd mention the possibility

Edit: and I also checked standard PROD5, they also don't use CEFFIC
